lib/src/dbs/capabilities.rs

```diff
@@ -141,33 +177,65 @@ impl<T: Target + Hash + Eq + PartialEq + std::fmt::Debug> Targets<T> {
 #[derive(Debug, Clone)]
 pub struct Capabilities {
 	scripting: bool,
+	anon_access: bool,
 
 	allow_funcs: Arc<Targets<FuncTarget>>,
 	deny_funcs: Arc<Targets<FuncTarget>>,
 	allow_net: Arc<Targets<NetTarget>>,
 	deny_net: Arc<Targets<NetTarget>>,
 }
 
+impl std::fmt::Display for Capabilities {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(
+			f,
+			"scripting={}, anon_access={}, allow_funcs={}, deny_funcs={}, allow_net={}, deny_net={}",
+			self.scripting, self.anon_access, self.allow_funcs, self.deny_funcs, self.allow_net, self.deny_net
+		)
+	}
+}
+
 impl Default for Capabilities {
-	// By default, enable all capabilities
 	fn default() -> Self {
+		Self::new_allow_none()
+	}
+}
+
+impl Capabilities {
+	fn new_allow_none() -> Self {
+		Self {
+			scripting: false,
+			anon_access: false,
+
+			allow_funcs: Arc::new(Targets::None),
+			deny_funcs: Arc::new(Targets::None),
+			allow_net: Arc::new(Targets::None),
+			deny_net: Arc::new(Targets::None),
+		}
+	}
+
+	pub fn new_allow_all() -> Self {
 		Self {
 			scripting: true,
+			anon_access: true,
 
 			allow_funcs: Arc::new(Targets::All),
 			deny_funcs: Arc::new(Targets::None),
 			allow_net: Arc::new(Targets::All),
 			deny_net: Arc::new(Targets::None),
 		}
 	}
-}
 
-impl Capabilities {
 	pub fn with_scripting(mut self, scripting: bool) -> Self {
 		self.scripting = scripting;
 		self
 	}
 
+	pub fn with_anon_access(mut self, anon_access: bool) -> Self {
```

Comment:
How about `with_guest_access`? IMHO `guest` might be better here from an API perspective because it's a full word.

Reply:
I like `guest` better than `anonymous`. It denotes the purpose of the capability better.

Reply:
I would concur with this sentiment. I might use `allow_guest` here to be more in line with the other flag naming conventions.